build.gradle.kts

```diff
@@ -0,0 +1,162 @@
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+}
+
+kotlin {
+    explicitApi()
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+lateinit var sourcesJar: AbstractArchiveTask
+lateinit var javadocJar: AbstractArchiveTask
+
+tasks {
+
+    compileKotlin {
+        kotlinOptions {
+            jvmTarget = "1.8"
+            apiVersion = "1.3"
+            languageVersion = "1.4" // Can be read by 1.3 compiler, so consumers on kotlin 1.3 are still supported.
+            freeCompilerArgs = listOf(
+                "-opt-in=kotlin.RequiresOptIn" // Using RequiresOptIn requires opt-in itself, at least in kotlin-1.4
+            )
+        }
+    }
+
+    ktlint {
+        version.set("0.45.2")
+        outputToConsole.set(true)
+    }
+
+    sourcesJar = create<Jar>("sourcesJar") sourcesJar@{
+        archiveClassifier.set("sources")
+        from(sourceSets.named("main"))
+    }
+
+    javadocJar = create<Jar>("javadocJar") javadocJar@{
+        archiveClassifier.set("javadoc")
+        from(javadoc)
+    }
+
+    withType<Sign> {
+        setOnlyIf { isReleaseVersion && gradle.taskGraph.hasTask(":publish") }
+    }
+
+    test {
+        useJUnitPlatform()
+        // report is always generated after tests run
+        finalizedBy(jacocoTestReport)
+    }
+
+    jacocoTestReport {
+        dependsOn(test)
+        reports {
+            xml.required.set(true)
+            html.required.set(true)
+        }
+        doLast {
+            logger.quiet("Coverage report written to file://${reports.html.outputLocation.get()}/index.html")
+        }
+    }
+}
+
+publishing {
+    publications.create<MavenPublication>("ion-element") {
+        // from(components.java)
+        artifact(sourcesJar)
+        artifact(javadocJar)
+
+        pom {
+            name.set("Ion Element Kotlin")
+            description.set(project.description)
+            url.set(githubRepositoryUrl)
+            scm {
+                connection.set("scm:git:git@github.com:amazon-ion/ion-java.git")
+                developerConnection.set("scm:git:git@github.com:amazon-ion/ion-java.git")
+                url.set("git@github.com:amazon-ion/ion-java.git")
+            }
+            licenses {
+                license {
+                    name.set("The Apache License, Version 2.0")
+                    url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                }
+            }
+            developers {
+                developer {
+                    name.set("Amazon Ion Team")
+                    email.set("ion-team@amazon.com")
+                    organization.set("Amazon Ion")
+                    organizationUrl.set("https://github.com/amazon-ion")
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            url = uri("https://aws.oss.sonatype.org/service/local/staging/deploy/maven2")
```

Comment:
Is this right? For ion-java I had to use `https://aws.oss.sonatype.org/service/local/`